shell/plugins/codemirror.js

```diff
@@ -35,8 +31,22 @@ import 'codemirror/addon/hint/anyword-hint.js';
 
 import { strPad } from '@shell/utils/string';
 
-Vue.use(VueCodemirror);
-export default VueCodemirror;
+const vCodemirror = {
+  install: (Vue) => {
+    if (Vue.component('codemirror')) {
+      // eslint-disable-next-line no-console
+      console.debug('Skipping codemirror install. Component already exists.');
+    } else {
+      Vue.component('codemirror', vueCodemirror);
```

Comment:
If I understand correctly, this is essentially doing the same thing as the install function does for `vue-codemirror`[^1]

[^1]: https://github.com/surmon-china/vue-codemirror/blob/v4.0.6/src/index.js

I think this is fine, but I'm thinking that it might make sense to document this case to make it clear why we changed from `Vue.use()` to `Vue.component()`. What do you think?